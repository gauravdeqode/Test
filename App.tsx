/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import type {PropsWithChildren} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,TouchableHighlight,TouchableOpacity
} from 'react-native';
import {
  Colors,
  DebugInstructions,
  Header,
  LearnMoreLinks,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import Controller from './Controller'

type TodoProps={ todo:{todoText:string}, editTodoItem:Function, deleteTodoItem:Function }

export const TodoItem=({ todo, editTodoItem, deleteTodoItem }: TodoProps)=> {
  return (
    <>
        <View className="todoItemText">
          <Text>{todo.todoText}</Text>
        </View>
        <View className="todoItemControls">
          <TouchableHighlight
            className="bg-default"
            onPress={() => editTodoItem(todo)}
          >
            <Text>Edit</Text>
          </TouchableHighlight>
          <TouchableHighlight
            className="bg-danger"
            onPress={() => deleteTodoItem(todo)}
          >
            <Text>Del</Text>
          </TouchableHighlight>
        </View>
      {/* </View> */}
    </>
  );
}

export class MainComp extends Controller {
  
  constructor(props) {
    super(props);
    console.log('constructor of MainComp')
  }

   componentDidMount() {
    console.log('HI componentDidMount of MainComp')

 //   console.log('props......',this.props)
  }
  componentWillMount(): void {
    console.log('HI componentWillMount of MainComp')
  }
  // static getDerivedStateFromProps(){
  //   console.log('HI getDerivedStateFromProps of MainComp')
  // }
//   async componentWillUnmount() {
//     console.log('HI componentWillUnmount of MainComp')
//   }


  render(): React.ReactNode {
    console.log('render maincomp')
   let todoProps:TodoProps={
    editTodoItem:()=>{},
    todo:{ todoText: "go to church" },
  deleteTodoItem:()=>{},
   }
    return (
      <View>
        {TodoItem(todoProps)}
      </View>
    )
  }
}



function App(): React.JSX.Element {
  const backgroundStyle = {
  backgroundColor: Colors.darker
  };

  const MainCompProps={
    onPressBtn:()=>{

    }
  }

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        //barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      <MainComp {...MainCompProps}/>
    {/* <MainComp {...MainCompProps}/> */}
      <TouchableOpacity
      onPress={()=>{
        console.log('HOLA TESTING')
      }}
      >
      </TouchableOpacity>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;
