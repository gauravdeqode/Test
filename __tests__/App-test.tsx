/**
 * @format
 */

import 'react-native';
import React from 'react';
import App, { MainComp, TodoItem } from '../App';
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Note: test renderer must be required after react-native.
import renderer,{} from 'react-test-renderer';

const findElement =(treeArr,testID)=>{ 
  const isDefined = undefined;
  for(index in treeArr){
    if(treeArr[index].props.testId == testID){
      isDefined = true;
    }
  }
  return isDefined;
}

it('Identify element by testID',()=>{
  const tree = renderer.create(<MainComp/>).toJSON();
  expect(findElement(tree.children,'EmailTextInput')).toBeDefined()
})

// configure({ adapter: new Adapter() });

// const screenProps = {
//   navigation: undefined,
//   id: ''
// }

// test('renders without crashing', () => {
//   const a =shallow(<MainComp />);
//   console.log('jiooooooooooooooooooooooooooooooo',a)
// });

// test('renders MainComp components', () => {
//   const wrapper = shallow(<App />);
//   expect(wrapper.find('MainComp')).toHaveLength(2);
//   wrapper.find("TouchableOpacity").at(0).props().onPress()
// });

// test('checking components states and methods', () => {
//   const wrapper = shallow(<MainComp />);
//   const instance = wrapper.instance();
//   instance.setState({token:""})
//   instance.hola()
//   instance.componentDidMount()
//   instance.componentWillUnmount()
//   //instance.props.onPressBtn()
//   //instance.setState({enableSignUp: true})
//   // @ts-ignore
// });

// const repostAPI = new Message(
//   getName(MessageEnum.RestAPIResponceMessage)
// );
// repostAPI.addData(
//   getName(MessageEnum.RestAPIResponceDataMessage),
//   repostAPI.messageId
// );
// // const apiRequestCallId = message.getData(getName(MessageEnum.RestAPIResponceDataMessage))
// repostAPI.addData(
//   getName(MessageEnum.RestAPIResponceSuccessMessage),
//   responseJson
// );
// instance.getQuizzesListRequestId = repostAPI.messageId;
// runEngine.sendMessage("Unit Test", repostAPI);

