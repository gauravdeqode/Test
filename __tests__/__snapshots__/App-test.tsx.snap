// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <Text>
    HOLA c
  </Text>
</RCTSafeAreaView>
`;
